enarx-keep-sev-shim/src/hostcall.rs

```diff
@@ -136,14 +136,16 @@ impl<'a> HostCall<'a> {
     pub unsafe fn write(&mut self, fd: usize, bytes: &[u8]) -> Result<libc::c_int, libc::c_int> {
         let bytes = &bytes[..core::cmp::min(bytes.len(), Block::buf_capacity())];
 
-        let buf_address = Address::from(self.0.buf.as_mut_ptr());
-        let shim_virt_address = ShimVirtAddr::try_from(buf_address).map_err(|_| libc::EFAULT)?;
+        let mut cursor = self.0.cursor();
 
-        let phys = ShimPhysAddr::from(shim_virt_address);
-        let request = request!(libc::SYS_write => fd, phys, bytes.len());
+        let untrusted = match cursor.copy_slice(bytes).or(Err(libc::EMSGSIZE)) {
+            Ok(slice) => slice,
+            Err(e) => return Err(e),
+        };
```

Comment:
I suspect we can use `?` here instead of match. @connorkuehl ?

Reply:
@npmccallum @lkatalin correct

Reply:
I'll make that change. If it looks reasonable otherwise, then that's great.